src/util/yaksu_handle_pool.c

```diff
@@ -38,12 +38,18 @@ typedef struct handle {
     UT_hash_handle hh;
 } handle_s;
 
+#define HANDLE_CACHE_SIZE   (16384)
+
 typedef struct handle_pool {
     pthread_mutex_t mutex;
 
     yaksu_handle_t next_handle; /* next brand-new handle (never allocated) */
     handle_s *free_handles;     /* list of handles that were allocated, but later freed */
     handle_s *used_handles;     /* hashmap of handles in use */
+
+    /* the first few handles are stored in a direct access cache, so
+     * we can get to these objects without having to grab a lock */
+    handle_s *handle_cache[HANDLE_CACHE_SIZE];
 } handle_pool_s;
 
 static pthread_mutex_t global_mutex = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
It is not directly related to this PR, but do you need this global mutex?
In other words, if `global_mutex` is protecting global variables modified in macros, I should miss something important for reviewing this PR.

Reply:
The global mutex is for pool creation.  Right now we only have two handle pools -- type handles and request handles.  So you are right that we don't need a global mutex.  But in the next PR we would have more pools (e.g., when we add multiple contexts).